be/src/storage/lake/vacuum.cpp

```diff
@@ -302,7 +301,8 @@ static Status collect_files_to_vacuum(TabletManager* tablet_mgr, std::string_vie
                 if (metadata->has_commit_time() && metadata->commit_time() > 0) {
                     compare_time = metadata->commit_time();
                 } else {
-                    ASSIGN_OR_RETURN(compare_time, fs->get_file_modified_time(path));
+                    // This is no longer needed
+                    // ASSIGN_OR_RETURN(compare_time, fs->get_file_modified_time(path));
```

Comment:
this should be for compatibility case where the metadata was upgraded from an earlier version that the commit_time is not there in the tabletMeta protobuf. better keep this fallback branch and may want to make a INFO/WARN log if hits this branch.

Reply:
this can be compatible with upgrading from earlier version which has no commit_time. In that case, the compare_time is 0, making a result that the vacuum will keep latest 1 version, which is ok for this case.

Reply:
I don't think it is ok. e.g. several versions are just created with 30 mins grace period (with no commit time), and then the cluster is upgraded to this new version. Before the upgrade, their is no version to be garbaged, after the upgrade, only one version will be preserved, all other versions are cleaned.

It is a behavior change in the time window.

Reply:
yes, I think this behavior is ok.

Reply:
avoid unnecessary behavior change as much as possible. This change introduces no significant benefit while it introduces side effect. Doesn't worth to make the change.

Reply:
I agree what you said, but I haven't figured out a better way to maintain full compatible behavior, and the new behavior I think it's acceptable

Reply:
maybe I don't get the original problem here, if just keep this line unchanged, any issue for this creation optimization?

Reply:
In this creation optimization, the path of a tablet metadata is not predictable, `fs->get_file_modified_time(path)` needs the path of the tablet metadata, so I remove it.

Reply:
Now I see the problem, it is the `path` not available since the interface is changed to get the metadata directly without return the meta file path back.

First, it is doable to get backwards compatible since the version is still here and it is possible to get the path back from the version and tablet id.
Second, even it is doable to get the backwards compatibility, it may not worth to put the extra effort to keep compatibility here, since the incompatibility will be just vanished after a few versions compaction/GC.

So I guess, it would be good to give enough comments here to explain why we don't want to keep backwards compatibility and why, just in case there are cases that requires the backwards compatibility again, we can pick up what we thought here and rethink about the situation in future.

Reply:
> yes, I think this behavior is ok.

b.t.w. usually it is very subjective to give such explanation, it's so vague.

Reply:
added some comments